server/src/main/java/org/opensearch/index/mapper/FlatObjectFieldMapper.java

```diff
@@ -380,6 +394,23 @@ public String rewriteValue(String searchValueString) {
 
         }
 
+        private boolean hasDotPath(String input) {
```

Comment:
This method seems to be wrongly named - it never checks for `.` presence but rather checks if the field starts with `mappedFieldTypeName`, is it by design?

Reply:
right by design, that's checking a field starts with `mappedFieldTypeName`, which is technically indicating if a query is checking should query `._value` subfield or .`_valueAndPath` subfield.  how about changing the naming for `hasMappedFieldTyeNameInQueryFieldName` ?

Reply:
:+1: to `hasMappedFieldTyeNameInQueryFieldName`  

> which is technically indicating if a query is checking should query ._value subfield or ._valueAndPath subfield. 

this is just for own curiosity, could you please share the sample of the query that has that? (since it is used to rewrite the value, I had difficulties to craft the query that take this path), thank you

Reply:
sure. This[ term query test ](https://github.com/opensearch-project/OpenSearch/blob/3ec33621f9bbb38d5cbf56133397ec7c218d6dbd/server/src/test/java/org/opensearch/index/mapper/FlatObjectFieldTypeTests.java#L66C1-L66C1)cover how to subfield is directed. even though  this `hasMappedFieldTyeNameInQueryFieldName` is a private method and cannot be directed tested, it's tested in the public method `directSubfield`. Does this test make sense, if not, I can added new tests for directSubfield

Reply:
added tests for `directSubfield`[here](https://github.com/opensearch-project/OpenSearch/blob/e9a25c7758a67f148ab5939d85ad25916ec0de44/server/src/test/java/org/opensearch/index/mapper/FlatObjectFieldTypeTests.java#L55)  and 'rewriteValue'  [here](https://github.com/opensearch-project/OpenSearch/blob/e9a25c7758a67f148ab5939d85ad25916ec0de44/server/src/test/java/org/opensearch/index/mapper/FlatObjectFieldTypeTests.java#L75)